### PR TITLE
EZP-26686: Add eZ Matrix Field Type to the Platform as a supported bundle

### PR DESCRIPTION
**Pull request by Nattfarinn**
JIRA: https://jira.ez.no/browse/EZP-26686

### CODE REVIEW

File: src/bundle/Resources/config/services/components.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+services:
+    ezplatform.matrix.fieldtype.javascripts:
+        parent: EzSystems\EzPlatformAdminUi\Component\TwigComponent
+        autowire: true
+        autoconfigure: false
+        public: false
+        arguments:
+            $template: '@@ezdesign/matrix_fieldtype/javascripts.component.html.twig'
```
Comment (webhdx) [inline src/bundle/Resources/config/services/components.yml]: `javascripts_component.html.twig`? We use underscore in filenames.

File: src/bundle/Resources/schema/legacy_matrix.xsd [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!--
+  ~ @copyright Copyright (C) eZ Systems AS. All rights reserved.
+  ~ @license For full copyright and license information view LICENSE file distributed with this source code.
+  -->
+
+<!--
```
Comment (webhdx) [inline src/bundle/Resources/schema/legacy_matrix.xsd]: Nitpick: Multiple copyright section

File: src/lib/Form/Type/FieldType/MatrixFieldType.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            /** @var \EzSystems\EzPlatformMatrixFieldtype\FieldType\Value\Row $originalRow */
+            foreach ($value->getRows() as $originalRow) {
+                $cells = $originalRow->getCells();
+                $rows[] = new Row(array_intersect_key($cells, $columnsByIdentifier));
+            }
+
+            $value->setRows(new RowsCollection($rows ?? []));
+        });
+
+        $builder->addModelTransformer(new class() implements DataTransformerInterface {
```
Comment (webhdx) [inline src/lib/Form/Type/FieldType/MatrixFieldType.php]: Matko bosko... Can you extract it to a proper class?

File: src/lib/FieldType/Type.php [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+     *
+     * @return mixed
+     */
+    protected function getSortInfo(FieldTypeValue $value)
+    {
+        return '';
+    }
+
+    /**
+     * Validates the fieldSettings of a FieldDefinitionCreateStruct or FieldDefinitionUpdateStruct.
```
Comment (webhdx) [inline src/lib/FieldType/Type.php]: Adding `{@inheritdoc}`  in all methods from the interface would improve readability of this class greatly.

File: src/bundle/Resources/config/services/components.yml (hunk 1 above)
Comment (Nattfarinn) [reply src/bundle/Resources/config/services/components.yml]: This is the name we use in few places already. Blame frontend!

File: src/lib/FieldType/Type.php (hunk 4 above)
Comment (Nattfarinn) [reply src/lib/FieldType/Type.php]: Anonymous classes are part of PHP 7+ and you have to start accepting them... They have feelings too!

File: src/lib/Form/Type/FieldType/MatrixFieldType.php (hunk 3 above)
Comment (Nattfarinn) [reply src/lib/Form/Type/FieldType/MatrixFieldType.php]: Anonymous classes are part of PHP 7+ and you have to start accepting them... They have feelings too!

**Pull request merged**